packages/istanbul-report-modern-html/lib/index.js

```diff
@@ -0,0 +1,181 @@
+/*
+ Copyright 2012-2015, Yahoo Inc.
+ Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
+ */
+const fs = require('fs');
+const path = require('path');
+const HtmlReport = require('istanbul-reports/lib/html');
+
+const standardLinkMapper = {
+    getPath(node) {
+        if (typeof node === 'string') {
+            return node;
+        }
+        let filePath = node.getQualifiedName();
+        if (node.isSummary()) {
+            if (filePath !== '') {
+                filePath += '/index.html';
+            } else {
+                filePath = 'index.html';
+            }
+        } else {
+            filePath += '.html';
+        }
+        return filePath;
+    },
+
+    relativePath(source, target) {
+        const targetPath = this.getPath(target);
+        const sourcePath = path.dirname(this.getPath(source));
+        return path.relative(sourcePath, targetPath);
+    },
+
+    assetPath(node, name) {
+        return this.relativePath(this.getPath(node), name);
+    }
+};
+
+function ModernHtmlReport(opts) {
```

Comment:
Let's just use a `class` here?

Reply:
This is just following the existing pattern in other modules - so its up to the maintainers..

Reply:
The pattern is due to many of the modules pre-dating node.js 4.  We now require node.js 6 (soon to be node.js 8), so we've been working towards modernizing the code.  It's a process and we haven't gotten around to updating any of the existing reports.